processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -849,17 +849,27 @@ public Rowboat apply(@Nullable Rowboat input)
         tree = new RTree(2, new LinearGutmanSplitStrategy(0, 50, bitmapFactory), bitmapFactory);
       }
 
-      BitmapIndexSeeker[] bitmapIndexSeeker = new BitmapIndexSeeker[indexes.size()];
+      BitmapIndexHolder[] bitmapIndexHolder = new BitmapIndexHolder[indexes.size()];
       for (int j = 0; j < indexes.size(); j++) {
-        bitmapIndexSeeker[j] = indexes.get(j).getBitmapIndexSeeker(dimension);
+        bitmapIndexHolder[j] = indexes.get(j).getBitmapIndexHolder(dimension);
       }
-      for (String dimVal : IndexedIterable.create(dimVals)) {
+      DictIdSeeker[] dictIdConverter = new DictIdSeeker[indexes.size()];
+      for (int j = 0; j < indexes.size(); j++) {
+        IntBuffer dimConversion = dimConversions.get(j).get(dimension);
+        if(dimConversion != null) {
```

Comment:
I use eclipse and find the eclipse_formatting.xml is not useful, and i will try use IntelliJ.